sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformConfiguration.java

```diff
@@ -59,20 +63,25 @@ public void validate() {
     final String confluentSchemaRegSubject = this.getConfluentSchemaRegistrySubject();
 
     if (confluentSchemaRegUrl != null) {
-      assert confluentSchemaRegSubject != null
-          : "To read from Kafka, a schema must be provided directly or though Confluent "
-              + "Schema Registry. Make sure you are providing one of these parameters.";
+      checkArgument(
+          confluentSchemaRegSubject != null,
+          "To read from Kafka, a schema must be provided directly or though Confluent "
+              + "Schema Registry. Make sure you are providing one of these parameters.");
     } else if (dataFormat != null && dataFormat.equals("RAW")) {
-      assert inputSchema == null : "To read from Kafka in RAW format, you can't provide a schema.";
+      checkArgument(
+          inputSchema == null, "To read from Kafka in RAW format, you can't provide a schema.");
     } else if (dataFormat != null && dataFormat.equals("JSON")) {
-      assert inputSchema != null : "To read from Kafka in JSON format, you must provide a schema.";
+      checkArgument(
+          inputSchema != null, "To read from Kafka in JSON format, you must provide a schema.");
     } else if (dataFormat != null && dataFormat.equals("PROTO")) {
-      assert messageName != null
-          : "To read from Kafka in PROTO format, messageName must be provided.";
-      assert fileDescriptorPath != null || inputSchema != null
-          : "To read from Kafka in PROTO format, fileDescriptorPath or schema must be provided.";
+      checkArgument(
+          messageName != null, "To read from Kafka in PROTO format, messageName must be provided.");
+      checkArgument(
+          fileDescriptorPath != null || inputSchema != null,
+          "To read from Kafka in PROTO format, fileDescriptorPath or schema must be provided.");
     } else {
-      assert inputSchema != null : "To read from Kafka in AVRO format, you must provide a schema.";
+      checkArgument(
+          inputSchema != null, "To read from Kafka in AVRO format, you must provide a schema.");
```

Comment:
nit: for many of these, checkNotNull() would be more appropriate

Reply:
Good point. Let me refactor that and push before emrging